Using global config is impacting the behaviour of the tests

When running the tests with this as global config.
```toml
# /home/jonathan/.config/ruff/ruff.toml
# https://github.com/charliermarsh/ruff#reference
target-version = "py37"
select = ["ALL"]
ignore = [
    "ANN101", # missing-type-self
    "ANN102", # missing-type-cls
    "ANN401", # dynamically-typed-expression
    "D10",    # missing-docstring
    "DTZ003", # call-datetime-utcnow
    "E501",   # line-too-long
]
unfixable = [
    "ERA001", # commented-out-code
    "F401",   # unused-import
    "F841",   # unused-variable
    "T20",    # print-found
]

[flake8-bugbear] # https://github.com/charliermarsh/ruff#flake8-bugbear
extend-immutable-calls = ["fastapi.Body", "fastapi.Depends"]

[flake8-builtins] # https://github.com/charliermarsh/ruff#flake8-builtins
builtins-ignorelist = []

[flake8-pytest-style] # https://github.com/charliermarsh/ruff#flake8-pytest-style-pt
fixture-parentheses = false
mark-parentheses = false
parametrize-names-type = "csv"

[isort]
known-third-party = []

[pep8-naming] # https://github.com/charliermarsh/ruff#pep8-naming
classmethod-decorators = [
    "classmethod",
    "pydantic.validator",
    "sqlalchemy.orm.declared_attr",
]

[pydocstyle] # https://github.com/charliermarsh/ruff#pydocstyle
convention = "pep257"
```
Running `cargo test --all` fails with this error messages
```console
---- test_stdin_filename stdout ----
thread 'test_stdin_filename' panicked at 'assertion failed: `(left == right)`
  left: `"F401.py:1:1: INP001 File `F401.py` is part of an implicit namespace package. Add an `__init__.py`.\nF401.py:1:8: F401 `os` imported but unused\nFound 2 errors.\n1 potentially fixable with the --fix option.\n"`,
 right: `"F401.py:1:8: F401 `os` imported but unused\nFound 1 error.\n1 potentially fixable with the --fix option.\n"`', ruff_cli/tests/integration_test.rs:48:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_stdin_json stdout ----
thread 'test_stdin_json' panicked at 'assertion failed: `(left == right)`
  left: `"[\n  {\n    \"code\": \"INP001\",\n    \"message\": \"File `F401.py` is part of an implicit namespace package. Add an `__init__.py`.\",\n    \"fix\": null,\n    \"location\": {\n      \"row\": 1,\n      \"column\": 1\n    },\n    \"end_location\": {\n      \"row\": 1,\n      \"column\": 1\n    },\n    \"filename\": \"/home/jonathan/Documents/github/ruff/ruff_cli/F401.py\"\n  },\n  {\n    \"code\": \"F401\",\n    \"message\": \"`os` imported but unused\",\n    \"fix\": null,\n    \"location\": {\n      \"row\": 1,\n      \"column\": 8\n    },\n    \"end_location\": {\n      \"row\": 1,\n      \"column\": 10\n    },\n    \"filename\": \"/home/jonathan/Documents/github/ruff/ruff_cli/F401.py\"\n  }\n]\n"`,
 right: `"[\n  {\n    \"code\": \"F401\",\n    \"message\": \"`os` imported but unused\",\n    \"fix\": {\n      \"content\": \"\",\n      \"message\": \"Remove unused import: `os`\",\n      \"location\": {\n        \"row\": 1,\n        \"column\": 0\n      },\n      \"end_location\": {\n        \"row\": 2,\n        \"column\": 0\n      }\n    },\n    \"location\": {\n      \"row\": 1,\n      \"column\": 8\n    },\n    \"end_location\": {\n      \"row\": 1,\n      \"column\": 10\n    },\n    \"filename\": \"/home/jonathan/Documents/github/ruff/ruff_cli/F401.py\"\n  }\n]\n"`', ruff_cli/tests/integration_test.rs:70:5


failures:
    test_stdin_filename
    test_stdin_json

test result: FAILED. 7 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.08s

error: test failed, to rerun pass `-p ruff_cli --test integration_test`
```
<!--
Thank you for taking the time to report an issue! We're glad to have you involved with Ruff.

If you're filing a bug report, please consider including the following information:

- A minimal code snippet that reproduces the bug.
- The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
- The current Ruff settings (any relevant sections from your `pyproject.toml`).
- The current Ruff version (`ruff --version`).
-->

--- a/ruff_cli/tests/integration_test.rs
+++ b/ruff_cli/tests/integration_test.rs
@@ -14,7 +14,7 @@ const BIN_NAME: &str = "ruff";
 #[test]
 fn test_stdin_success() -> Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
-    cmd.args(["-", "--format", "text"])
+    cmd.args(["-", "--format", "text", "--isolated"])
         .write_stdin("")
         .assert()
         .success();
@@ -25,7 +25,7 @@ fn test_stdin_success() -> Result<()> {
 fn test_stdin_error() -> Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
     let output = cmd
-        .args(["-", "--format", "text"])
+        .args(["-", "--format", "text", "--isolated"])
         .write_stdin("import os\n")
         .assert()
         .failure();
@@ -41,7 +41,14 @@ fn test_stdin_error() -> Result<()> {
 fn test_stdin_filename() -> Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
     let output = cmd
-        .args(["-", "--format", "text", "--stdin-filename", "F401.py"])
+        .args([
+            "-",
+            "--format",
+            "text",
+            "--stdin-filename",
+            "F401.py",
+            "--isolated",
+        ])
         .write_stdin("import os\n")
         .assert()
         .failure();
@@ -58,7 +65,14 @@ fn test_stdin_filename() -> Result<()> {
 fn test_stdin_json() -> Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
     let output = cmd
-        .args(["-", "--format", "json", "--stdin-filename", "F401.py"])
+        .args([
+            "-",
+            "--format",
+            "json",
+            "--stdin-filename",
+            "F401.py",
+            "--isolated",
+        ])
         .write_stdin("import os\n")
         .assert()
         .failure();
@@ -107,7 +121,7 @@ fn test_stdin_json() -> Result<()> {
 fn test_stdin_autofix() -> Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
     let output = cmd
-        .args(["-", "--format", "text", "--fix"])
+        .args(["-", "--format", "text", "--fix", "--isolated"])
         .write_stdin("import os\nimport sys\n\nprint(sys.version)\n")
         .assert()
         .success();
@@ -122,7 +136,7 @@ fn test_stdin_autofix() -> Result<()> {
 fn test_stdin_autofix_when_not_fixable_should_still_print_contents() -> Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
     let output = cmd
-        .args(["-", "--format", "text", "--fix"])
+        .args(["-", "--format", "text", "--fix", "--isolated"])
         .write_stdin("import os\nimport sys\n\nif (1, 2):\n     print(sys.version)\n")
         .assert()
         .failure();
@@ -137,7 +151,7 @@ fn test_stdin_autofix_when_not_fixable_should_still_print_contents() -> Result<(
 fn test_stdin_autofix_when_no_issues_should_still_print_contents() -> Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
     let output = cmd
-        .args(["-", "--format", "text", "--fix"])
+        .args(["-", "--format", "text", "--fix", "--isolated"])
         .write_stdin("import sys\n\nprint(sys.version)\n")
         .assert()
         .success();
@@ -152,7 +166,7 @@ fn test_stdin_autofix_when_no_issues_should_still_print_contents() -> Result<()>
 fn test_show_source() -> Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
     let output = cmd
-        .args(["-", "--format", "text", "--show-source"])
+        .args(["-", "--format", "text", "--show-source", "--isolated"])
         .write_stdin("l = 1")
         .assert()
         .failure();

